iceoryx_examples/automotive_soa/include/owl/kom/event_publisher_uds.inl

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_UDS_INL
+
+#include "owl/kom/event_publisher_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventPublisherUds<T>::EventPublisherUds(const core::String& service,
+                                               const core::String& instance,
+                                               const core::String& event) noexcept
+    : m_publisher({service, instance, event}, {1U, iox::NodeName_t(), true})
+    , m_instanceId(instance)
+{
+}
+
+template <typename T>
+inline std::unique_ptr<T> EventPublisherUds<T>::Allocate() noexcept
+{
+    // The proxy needs some time to discover the service and create the EventSubscriberUds with the UDS server,
+    // hence the creation of the UDS client is done here
+    if (!m_calledForTheFirstTime)
+    {
+        m_uds = std::move(iox::posix::UnixDomainSocket::create(m_instanceId, iox::posix::IpcChannelSide::CLIENT)
+                              .or_else([](auto&) {
+                                  std::cout << "Failed to create UNIX domain socket!" << std::endl;
+                                  std::terminate();
+                              })
+                              .value());
+        m_calledForTheFirstTime = true;
+    }
+
+    // Allocate the memory on the heap
+    return std::make_unique<SampleType>();
+}
+
+template <typename T>
+void EventPublisherUds<T>::Send(std::unique_ptr<SampleType> userSamplePtr) noexcept
+{
```

Comment:
After reviewing this method I came to the conclusion that you should follow the advice from here: https://github.com/eclipse-iceoryx/iceoryx/pull/1355/files#r876024193

and perform a simple `memcpy` otherwise you will open a complex serialization which should not be part of such an example.